connectors/sources/dropbox.py

```diff
@@ -5,22 +5,38 @@
 #
 """Dropbox source module responsible to fetch documents from Dropbox online.
 """
-from functools import cached_property
+import asyncio
+import os
+from functools import cached_property, partial
 
+import aiofiles
 import dropbox
 import requests
+from aiofiles.os import remove
+from aiofiles.tempfile import NamedTemporaryFile
 from dropbox.exceptions import ApiError, AuthError, BadInputError
+from dropbox.files import FileMetadata
+from dropbox.sharing import FileLinkMetadata, SharedFileMetadata
 
 from connectors.logger import logger
 from connectors.source import BaseDataSource, ConfigurableFieldValueError
-from connectors.utils import CancellableSleeps, iso_utc
+from connectors.utils import (
+    TIKA_SUPPORTED_FILETYPES,
+    CancellableSleeps,
+    convert_to_b64,
+    iso_utc,
+)
 
 RETRY_COUNT = 3
+FILE_SIZE_LIMIT = 10485760  # Size in Bytes
```

Comment:
How did you come up with 10MB limit?

Reply:
The ElasticSearch is having the default value for indexing a file of 10 MB and so we've kept this as a constant.